### ISSUE TITLE
refresh() resets editor position when string ends with certain whitespace characters

### ISSUE THREAD

**Issue by jobh** [josdejong/jsoneditor#1307]
To implement interactive display of differences, like in jsoneditoronline, I call `refresh()` after every change, so that `onClassName` is called again and has a chance to modify the css. The problem is that this `refresh` resets the insert position every time I add a space (or certain other) character to a string value.

(Note: This happens whether or not onClassName is set. I have tested with version 9.3.1 only.)

To reproduce, change `options` as follows in the sample in README.md:
```
        const options = { onChange: () => editor.refresh() }
```
and then edit the string value to add a space or a backslash at the end. This may also affect other characters.

**Comment by josdejong** [comment]
Thanks for reporting, fixed via d633ee1fe40d9ef7855c10ddbc4500aea8fe8208 (it's a beautifully stupid bug, have a look at the commit 😉  )

Fixed now in `v9.4.0`

**Comment by jobh** [comment]
Nice, thanks for the quick response!
